crates/pathfinder/src/state/sync.rs

```diff
@@ -871,57 +890,135 @@ async fn download_verify_and_insert_missing_classes<
 
     // For each missing, download, verify and insert definition.
     for class_hash in missing {
-        let definition = sequencer
-            .class_by_hash(class_hash)
-            .await
-            .with_context(|| format!("Downloading class {}", class_hash.0))?;
-
-        // Calculate the class hash. This can be expensive, so perform in a blocking task.
-        let extract = tokio::task::spawn_blocking(move || -> anyhow::Result<_> {
-            let hash = compute_class_hash(&definition)?;
-            Ok((definition, hash))
-        });
-        let (definition, hash) = extract
-            .await
-            .context("Parse class definition and compute hash")??;
+        let class = download_class(&sequencer, class_hash).await?;
+
+        match class {
+            DownloadedClass::Cairo(class) => {
+                tokio::task::block_in_place(|| {
```

Comment:
Why `block_in_place`? Should it not use `spawn_blocking`? Simiarly below.

Reply:
In fact, `block_in_place` is only used within this `sync.rs` file. I wonder if this is not something I did incorrectly when we first created this file.

Reply:
I don't think it's incorrect use. It's maybe slightly less efficient than using `spawn_blocking` for cases where we could easily use `spawn_blocking`. There are a _lot_ of these in `sync.rs`, so maybe handling those in a separate PR is OK?

Reply:
I suppose we may as well just wait for most of this to get replaced for p2p.